wallet/wallet.h

```diff
@@ -684,4 +688,26 @@ void wallet_htlc_sigs_save(struct wallet *w, u64 channel_id,
 bool wallet_network_check(struct wallet *w,
 			  const struct chainparams *chainparams);
 
+/**
+ * wallet_block_add - Add a block to the blockchain tracked by this wallet
+ */
+void wallet_block_add(struct wallet *w, struct block *b);
+
+/**
+ * wallet_block_remove - Remove a block (and all its descendants) from the tracked blockchain
```

Comment:
Comment is no longer accurate --- now it only removes a block without affecting its descendants if any.

Looking over the rest of the code it seems `wallet_block_remove` is only called on the current tip? If so, suggest adding an assertion somewhere in the implementation of `wallet_block_remove`.

    /* after DELETE */
    stmt = db_prepare(w->db, "SELECT * FROM blocks WHERE height >= ?;");
    sqlite3_bind_int(stmt, 1, b->height);
    assert(sqlite3_step(stmt) == SQLITE_DONE);
    sqlite3_finalize(stmt);